doc/develop-plugin-csharp.md

```diff
@@ -0,0 +1,103 @@
+
+Note: This file is not necessary if you're extending an existing plugin for Tizen, so you can safely remove it.
+
+### 2. Add Tizen platform code (`.cs`)
+
+The implementation of the plugin can be found in the `tizen/PluginNamePlugin.cs` file. This file has a `public` class named `PluginNamePlugin` that implements the `IFlutterPlugin` interface. The `IFlutterPlugin` interface is used to handle the lifecycle of the plugin.
+
+- `OnAttachedToEngine(IFlutterPluginBinding binding)`: This method is called when the plugin is registered to the `DotnetPluginRegistry` by the flutter application. In this method, you can create and initialize the platform channel.
+
+```c#
+        public void OnAttachedToEngine(IFlutterPluginBinding binding)
+        {
+            _channel = new MethodChannel("plugin_name", StandardMethodCodec.Instance, binding.BinaryMessenger);
+            _channel.SetMethodCallHandler(HandleMethodCall);
+        }
+```
+
+- `OnDetachedFromEngine()`: This method is called when the plugin is unregistered from the `DotnetPluginRegistry` by the flutter application. You can release the resources allocated by the platform channel in this method.
+
+```c#
+        public void OnDetachedFromEngine()
+        {
+            _channel.UnsetMethodCallHandler();
+            _channel = null;
+        }
+```
+
+In case of using `MethodChannel` to communicate with the application, you can set a callback method to handle the method call using `SetMethodCallHandler` method. The callback method can be synchronous or asynchronous.
+
+```c#
+        // Handle method calls on the plugin's channel.
+        private object HandleMethodCall(MethodCall call) {
+            if (call.Method == "getPlatformVersion") {
+                return Foo.GetPlatformVersion();
+            }
+            throw new MissingPluginException();
+        }
+
+        // Handle method calls on the plugin's channel asynchronously.
+        private async Task<object> HandleMethodCallAsync(MethodCall call) {
+            if (call.Method == "getPlatformVersion") {
+                return await Foo.GetPlatformVersionAsync();
+            }
+            throw new MissingPluginException();
+        }
+```
+
+In the callback method, you can use the `MethodCall` object to get the method name and arguments. If the method is completed successfully, just return any value to the application. If the method is failed, throw an exception to the application. The exception will be caught by the `MethodChannel` and the error message will be returned to the application. If the method is not implemented, throw a `MissingPluginException` to the application.
```

Comment:
> If the method is failed, throw an exception to the application.

- "is failed" → "has failed" or "failed"
- Which type of exception should be thrown by the plugin?

Reply:
> Any exception type can be thrown, but `FlutterException` must be used to specify the code and details of the error.

Any exception type can be used but `FlutterException` must be used? Sounds like a contradiction.

Reply:
how about .. 
>  Any exception type can be thrown. However, if you want to specify the error code and details, use `FlutterException`.

?

Reply:
Looks good :)

Reply:
Done. :smile_cat: